### PR TITLE
Update libgit2 to a56db99

### PR DESCRIPTION
**Pull request by nulltoken**
https://github.com/libgit2/libgit2/compare/ff8d635...a56db99

### CODE REVIEW

File: LibGit2Sharp/Core/GitCredentialType.cs [hunk 1]
```diff
@@ -32,5 +32,20 @@ internal enum GitCredentialType
...
         /// </summary>
         SshInteractive = (1 << 4),
+
+        /// <summary>
+        /// Username-only information
+        ///
+        /// If the SSH transport does not know which username to use,
+        /// it will ask via this credential type.
+        /// </summary>
+        Username = (1 << 5),
```
Comment (Therzok) [inline LibGit2Sharp/Core/GitCredentialType.cs]: whyyyy?

Comment (nulltoken) [inline LibGit2Sharp/Core/GitCredentialType.cs]: Huh? As you know, I'm not very good at ssh-thingies

Comment (Therzok) [inline LibGit2Sharp/Core/GitCredentialType.cs]: I meant why add it? I can add the ssh-memory stuff in my PR, just don't want merge conflicts. :(

Comment (Therzok) [inline LibGit2Sharp/Core/GitCredentialType.cs]: Nevermind. :D

**Pull request merged**